service/service.go

```diff
@@ -93,60 +95,81 @@ func New(config Config) (*Service, error) {
 		return nil, microerror.Mask(err)
 	}
 
-	var awsClusterConfigFramework *framework.Framework
+	var healthzService *healthz.Service
 	{
-		baseClusterConfig, err := newBaseClusterConfig(config.Flag, config.Viper)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-
-		c := awsclusterconfig.FrameworkConfig{
-			BaseClusterConfig: baseClusterConfig,
-			G8sClient:         g8sClient,
-			K8sClient:         k8sClient,
-			K8sExtClient:      k8sExtClient,
-
-			Logger:      config.Logger,
-			ProjectName: config.ProjectName,
+		c := healthz.Config{
+			K8sClient: k8sClient,
+			Logger:    config.Logger,
 		}
 
-		awsClusterConfigFramework, err = awsclusterconfig.NewFramework(c)
+		healthzService, err = healthz.New(c)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 	}
 
+	var providerKind string
+	var awsClusterConfigFramework *framework.Framework
 	var kvmClusterConfigFramework *framework.Framework
 	{
 		baseClusterConfig, err := newBaseClusterConfig(config.Flag, config.Viper)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 
-		c := kvmclusterconfig.FrameworkConfig{
-			BaseClusterConfig: baseClusterConfig,
-			G8sClient:         g8sClient,
-			K8sClient:         k8sClient,
-			K8sExtClient:      k8sExtClient,
-
-			Logger:      config.Logger,
-			ProjectName: config.ProjectName,
-		}
-
-		kvmClusterConfigFramework, err = kvmclusterconfig.NewFramework(c)
-		if err != nil {
-			return nil, microerror.Mask(err)
+		providerKind = config.Viper.GetString(config.Flag.Service.Provider.Kind)
+		providerKind = strings.ToLower(strings.TrimSpace(providerKind))
+		switch providerKind {
+		case "aws":
```

Comment:
I think constants for the provider kinds is better as they are used in a few places.